test/Soenneker.Utils.AutoBogus.Tests/AutoFakerTests.cs

```diff
@@ -349,6 +353,65 @@ public void Generate_ImmutableArray_should_generate()
         immutableListDto.Array.Should().NotBeNullOrEmpty();
     }
 
+    
+    public sealed class ExampleClass
+    {
+        public string? Value { get; init; }
+        public string AlwaysSet { get; init; }
+    }
+
+    [Fact]
+    public void Generate_RulesSet_Should_Generate_Wtih_Ruleset_And_Oeverride()
```

Comment:
This is just or my testing, not sure how to name a test that checks for this issue I would go with

`Genrate_ShouldUserCorrectRuleset_WhenRulestChangesInSubsiquentCalles`

But that is a different style